From 57a1a449b7305604710db32ffd9ab413492e1f5b Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Sun, 24 Dec 2023 17:23:39 +0100
Subject: [PATCH 1/7] Fix overwriting INCLUDE_DIRECTORIES variable

Let's set it before adding more include directories. Fixes compilation
on Alpine Linux as of day - with CMake 3.28.1 fwiw
---
 tests/uqmlscene/CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/tests/uqmlscene/CMakeLists.txt b/tests/uqmlscene/CMakeLists.txt
index 6cffcc4f1..6c0cce047 100644
--- a/tests/uqmlscene/CMakeLists.txt
+++ b/tests/uqmlscene/CMakeLists.txt
@@ -6,6 +6,10 @@ add_executable(uqmlscene
 
 pkg_check_modules(XCB REQUIRED xcb)
 
+if (NOT "${XCB_INCLUDE_DIRS}" STREQUAL "")
+    set_target_properties(uqmlscene PROPERTIES INCLUDE_DIRECTORIES ${XCB_INCLUDE_DIRS})
+endif()
+
 include_directories(
     SYSTEM
     ${Qt5Quick_PRIVATE_INCLUDE_DIRS}
@@ -19,10 +23,6 @@ include_directories(
     ${liblomiri-private_SOURCE_DIR}
 )
 
-if (NOT "${XCB_INCLUDE_DIRS}" STREQUAL "")
-    set_target_properties(uqmlscene PROPERTIES INCLUDE_DIRECTORIES ${XCB_INCLUDE_DIRS})
-endif()
-
 target_link_libraries(uqmlscene
     Qt5::Qml Qt5::Quick Qt5::Test Qt5::DBus
     ${Qt5Qml_LIBRARIES}
-- 
2.43.0

